### PR TITLE
fix: safely completing doClose()

### PR DESCRIPTION
**Pull request by gtassone**
Ensure that `doClose()` completes and successfully transitions the StreamProcessor to CLOSED state. Unhandled exception during `doClose()` results in a degraded StreamProcessor with terminated control loop but stuck in RUNNING state. 

This is critical to support accurate external monitoring of `isClosedOrFailed()` by user code.

see https://github.com/confluentinc/parallel-consumer/issues/809

### Checklist

- [ ] Documentation (if applicable)
- [ ] Changelog

### CODE REVIEW

File: .sdkmanrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Enable auto-env through the sdkman_auto_env config
```
Comment (rkolesnev) [inline .sdkmanrc:1]: could you please remove this file from check-in / add to your local gitignore?

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 2]
```diff
@@ -678,23 +695,26 @@ private void doClose(Duration timeout) throws TimeoutException, ExecutionExcepti
         if (Thread.currentThread().isInterrupted()) {
             log.warn("control thread interrupted - may lead to issues with transactional commit lock acquisition");
         }
-        commitOffsetsThatAreReady();
-
+        try {
+            commitOffsetsThatAreReady();
+        } catch (Exception e) {
+            log.error("failed to commit during close sequence", e);
```
Comment (rkolesnev) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:701]: I think warning is more warranted here? there should be already error logged that lead to close() invocation.
```suggestion
            log.warn("failed to commit during close sequence", e);
```

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 3]
```diff
@@ -678,23 +695,26 @@ private void doClose(Duration timeout) throws TimeoutException, ExecutionExcepti
...
+        }
         // only close consumer once producer has committed it's offsets (tx'l)
         log.debug("Closing and waiting for broker poll system...");
-        brokerPollSubsystem.closeAndWait();
+        try {
+            brokerPollSubsystem.closeAndWait();
+        } catch (Exception e) {
+            log.error("failed to close brokerPollSubsystem during close sequence", e);
```
Comment (rkolesnev) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:708]:
```suggestion
            log.warn("failed to close brokerPollSubsystem during close sequence", e);
```

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 4]
```diff
@@ -678,23 +695,26 @@ private void doClose(Duration timeout) throws TimeoutException, ExecutionExcepti
...
+        }
 
-        maybeCloseConsumer();
+        try {
+            maybeCloseConsumer();
+        } catch (Exception e) {
+            log.error("failed to maybeCloseConsumer during close sequence", e);
```
Comment (rkolesnev) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:714]:
```suggestion
            log.warn("failed to maybeCloseConsumer during close sequence", e);
```

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java [hunk 5]
```diff
@@ -629,6 +629,23 @@ private void waitForClose(Duration timeout) throws TimeoutException, ExecutionEx
     }
 
     private void doClose(Duration timeout) throws TimeoutException, ExecutionException, InterruptedException {
+        try {
```
Comment (rkolesnev) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:632]: could you please add a comment referencing github issue - and maybe a really short explanation for all the try / catch - that this is due to missing timeout / controlled shutdown on the poller / commit subsystems.

**Review by rkolesnev — CHANGES_REQUESTED** [review]

File: .sdkmanrc (hunk 1 above)
Comment (gtassone) [reply .sdkmanrc]: removed

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java (hunk 5 above)
Comment (gtassone) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java:634]: added comment

**Review by rkolesnev — APPROVED** [review]

**Pull request merged**